azurerm/resource_arm_action_group_test.go

```diff
@@ -0,0 +1,242 @@
+					resource.TestCheckResourceAttr(resourceName, "email_receiver.1.email_address", "devops@contoso.com"),
+					resource.TestCheckResourceAttr(resourceName, "sms_receiver.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "sms_receiver.0.country_code", "1"),
+					resource.TestCheckResourceAttr(resourceName, "sms_receiver.0.phone_number", "1231231234"),
+					resource.TestCheckResourceAttr(resourceName, "webhook_receiver.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "webhook_receiver.0.service_uri", "http://example.com/alert"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAzureRMActionGroup_empty(t *testing.T) {
+	resourceName := "azurerm_action_group.test"
+	ri := acctest.RandInt()
+	config := testAccAzureRMActionGroup_empty(ri, testLocation())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMActionGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMActionGroupExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "enabled", "true"),
+					resource.TestCheckResourceAttr(resourceName, "email_receiver.#", "0"),
+					resource.TestCheckResourceAttr(resourceName, "sms_receiver.#", "0"),
+					resource.TestCheckResourceAttr(resourceName, "webhook_receiver.#", "0"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAzureRMActionGroup_disabledEmpty(t *testing.T) {
+	resourceName := "azurerm_action_group.test"
+	ri := acctest.RandInt()
+	location := testLocation()
+	preConfig := testAccAzureRMActionGroup_disabledEmpty(ri, location)
+	postConfig := testAccAzureRMActionGroup_empty(ri, location)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMActionGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: preConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMActionGroupExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "enabled", "false"),
+				),
+			},
+			{
+				Config: postConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMActionGroupExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "enabled", "true"),
+				),
+			},
+			{
+				Config: preConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMActionGroupExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "enabled", "false"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
```

Comment:
in general we don't test importing the updates, but rather full configurations. in this case can we split this test into two: one to test with it in the disabled state, and one to ensure it can go from disabled -> enabled -> disabled